Task3.rb

```diff
@@ -0,0 +1,14 @@
+require 'benchmark'
+require "Prime"
+num = 600851475143
+prime = 0
+large_prime = 0
+while num != 1 do
+  prime = Prime.detect { |prime| num % prime == 0 }
+  num = num / prime
+  if prime > large_prime
+    large_prime = prime
+  end
+end
+puts large_prime
+puts Benchmark.measure { "a"*1_000_000_000 }
```

Comment:
Solution:
```
def largest_prime( n )
  divisor = 2

  while(true)
    while n % divisor == 0
      n /= divisor

      return divisor if n == 1
    end

    divisor += 1
  end
end
```
You can compare benchmarks and try to update your code. It's an optional action.